docker/Dockerfile.cu121.hf

```diff
@@ -0,0 +1,30 @@
+# NOTE: This Dockerfile is based on CUDA 12.1.
+# To benchmark on other CUDA versions, search and replace "12.1" and "121".
+FROM nvidia/cuda:12.1.0-devel-ubuntu22.04
+
+SHELL ["/bin/bash", "-ec"]
+
+# Step 1. Set up Ubuntu
+RUN grep -v '[ -z "\$PS1" ] && return' ~/.bashrc >/tmp/bashrc                                               && \
+    mv /tmp/bashrc ~/.bashrc                                                                                && \
+    echo "export HF_HOME=/hf/" >>~/.bashrc                                                                  && \
+    echo "export PATH=/usr/local/cuda/bin/:\$PATH" >>~/.bashrc                                              && \
+    apt update                                                                                              && \
+    apt install --yes wget curl git vim build-essential openssh-server
+
+# Step 2. Set up python
+RUN bash <(curl -L micro.mamba.pm/install.sh) && source ~/.bashrc                                           && \
+    micromamba create --yes -n python311 -c conda-forge                                                        \
+    python=3.11                                                                                             && \
+    micromamba activate python311                                                                           && \
+    pip install transformers==4.33.3 torch==2.0.1 accelerate==0.23.0 bitsandbytes==0.41.1 scipy==1.11.3
+
+# Step 3. Copy the benchmark script
+COPY benchmark_hf.py /root/benchmark_hf.py
+
+# Step 4. Set up SSH and clean up
+COPY install/ssh.sh /install/ssh.sh
+RUN bash /install/ssh.sh
+WORKDIR /root
+EXPOSE 22
+CMD ["/usr/sbin/sshd", "-D"]
```

Comment:
I suppose we don't need SSH any more given we could directly sign in into the docker image